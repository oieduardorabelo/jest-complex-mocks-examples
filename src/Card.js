import React from "react";

function Card(props) {
  return (
    <div>
      <p>Hello Card!</p>
      <button onClick={props.onClick}>Testing</button>
    </div>
  );
}

export default Card;
