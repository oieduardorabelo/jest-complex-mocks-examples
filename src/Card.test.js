import React from "react";
import { render, mount } from "enzyme";
import toJson from "enzyme-to-json";

import Card from "./Card";

it("renders correctly", () => {
  const actual = toJson(render(<Card />));
  expect(actual).toMatchSnapshot();
});

it("should call spy on click", () => {
  const onClickSpy = jest.fn();
  const actual = mount(<Card onClick={onClickSpy} />);
  actual.find("button").simulate("click");
  expect(onClickSpy).toHaveBeenCalledTimes(1);
});
