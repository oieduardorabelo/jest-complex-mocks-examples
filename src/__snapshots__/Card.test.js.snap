// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Hello Card!
  </p>
  <p>
    Card I18N: 
    <span>
      Today
    </span>
  </p>
  <button>
    Card Redux Action
  </button>
</div>
`;
